app/controllers/register/company/AreYouInUKController.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register.company
+
+import config.FrontendAppConfig
+import connectors.UserAnswersCacheConnector
+import controllers.actions._
+import forms.register.AreYouInUKFormProvider
+import forms.register.company.CompanyDetailsFormProvider
+import identifiers.register.company.{AreYouInUKId, CompanyDetailsId}
+import javax.inject.Inject
+import models.{Mode, NormalMode}
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils.annotations.RegisterCompany
+import utils.{Navigator, UserAnswers}
+import views.html.register.company.areYouInUK
+
+import scala.concurrent.Future
+
+class AreYouInUKController @Inject()(
+                                          appConfig: FrontendAppConfig,
+                                          override val messagesApi: MessagesApi,
+                                          dataCacheConnector: UserAnswersCacheConnector,
+                                          @RegisterCompany navigator: Navigator,
+                                          authenticate: AuthAction,
+                                          getData: DataRetrievalAction,
+                                          requireData: DataRequiredAction,
+                                          formProvider: AreYouInUKFormProvider
+                                        ) extends FrontendController with I18nSupport {
+
+  private val form = formProvider()
+
+  def onPageLoad(): Action[AnyContent] = (authenticate andThen getData andThen requireData) {
+    implicit request =>
+      val preparedForm = request.userAnswers.get(AreYouInUKId) match {
```

Comment:
I'd probably do this more concisely like this: request.userAnswers.get(AreYouInUKId).fold(form)(v=>form.fill(v))
However, it's a matter of subjective preference so no need to do that unless you agree.